libs/platform/src/lib/components/form/step-input/base.step-input.ts

```diff
@@ -0,0 +1,477 @@
+
+        if (needToUpdateView) {
+            this._updateViewValue();
+        }
+
+        if (this._value !== this.lastEmittedValue) {
+            this._emitChangedValue();
+        }
+    }
+
+    /**@hidden
+     * Indicates when input gets focused
+     */
+    onFocus(): void {
+        super._onFocusChanged(true);
+        // When focus happend by "Tab" key an input field value gets selected by default.
+        // In cases when new_formatted_value !== previous_value the selection gets lost.
+        // So it's needed to track selection before new value rendering and restore it.
+        const { selectionStart, selectionEnd } = this._getValueSelection();
+        this._updateViewValue();
+        if (selectionEnd - selectionStart > 0) {
+            this._setValueSelection(selectionStart, selectionEnd);
+        }
+    }
+
+    /**@hidden
+     * Indicates when input loses focus
+     */
+    onBlur(): void {
+        super._onFocusChanged(false);
+        this._updateViewValue();
+    }
+
+    /** @hidden */
+    detectChanges(): void {
+        this._cd.detectChanges();
+    }
+
+    /** Create valueChange event */
+    abstract createChangeEvent(value: number): StepInputChangeEvent;
+
+    /** Format value for view presentation */
+    abstract formatValue(value: number): string;
+
+    /** Format value for "in focus" mode */
+    abstract formatValueInFocusMode(value: number): string;
+
+    /** Parse value entered "in focus" mode */
+    abstract parseValueInFocusMode(value: string): number | null;
+
+    /** @hidden */
+    private _listenToFormErrorState(): void {
+        this.stateChanges
+            .asObservable()
+            .pipe(takeUntil(this._destroyed))
+            .subscribe(() => {
+                this.isErrorState = this.status === 'error';
+            });
+    }
+
+    /** @hidden */
+    private _listenToAlign(): void {
+        this._align$
+            .asObservable()
+            .pipe(
+                switchMap((align) =>
+                    this._rtlService.rtl.pipe(
+                        map(
+                            (isRtl): StepInputAlign => {
+                                if (!ALIGN_INPUT_OPTIONS_LIST.includes(align)) {
+                                    return null;
+                                }
+                                if (isRtl && align === StepInputAlign.Left) {
+                                    return StepInputAlign.Right;
+                                }
+                                if (isRtl && align === StepInputAlign.Right) {
+                                    return StepInputAlign.Left;
+                                }
+                                return align;
+                            }
+                        )
+                    )
+                ),
+                takeUntil(this._destroyed)
+            )
+            .subscribe((align) => {
+                this._align = align;
+                this.detectChanges();
```

Comment:
again here. Let's correctly figureout when to emit event to formField, when to markForCheck and when to call detectChanges.

Reply:
Changed it to `markForCheck()`.
We need `markForCheck` here because `rtlService` is out of the component context.